tests/Fixtures/SubscriptionRepositorySpy.php

```diff
@@ -26,4 +26,14 @@ public function getSubscriptions(): array {
 		return $this->subscriptions;
 	}
 
+	public function findByConfirmationCode( string $confirmationCode )
+	{
+		foreach( $this->subscriptions as $subscription ) {
+			if ( $subscription->getConfirmationCode() === $confirmationCode ) {
+				return $subscription;
+			}
+		}
```

Comment:
You are creating a "fake" here, something like an `InMemorySubscriptionRepository`. Spies just record the calls made.